### PR TITLE
do not allow non-database owners to make extensions

### PR DESCRIPTION
**Pull request by kvlou**
Continuation of what @genslein  did.

### CODE REVIEW

File: pgextwlist.c [hunk 1]
```diff
@@ -360,9 +360,29 @@ extwlist_ProcessUtility(PROCESS_UTILITY_PROTO_ARGS)
...
-										NULL, NULL, NULL, NULL, NULL);
-					return;
+					char *current_user = get_current_user_name();
+					char *database_owner = get_current_database_owner_name();
+					
+					if(current_user == NULL || database_owner == NULL)
+					{
+						ereport(ERROR,
+							(errcode(ERRCODE_INTERNAL_ERROR),
+							errmsg("Invalid current user (%s) or database owner (%s)", current_user, database_owner)));
```
Comment (genslein) [inline pgextwlist.c]: I'd say we don't print the DB owner name in case it's not null so that it's not logged plaintext style

**Review by genslein — APPROVED** [review]
My comment is non-blocking but I think a better practice

File: pgextwlist.c (hunk 1 above)
Comment (kvlou) [reply pgextwlist.c]: should that be a secret?:)

Comment (genslein) [reply pgextwlist.c]: this logic lets you send null as the current user and potentially would log the DB owner user. Maybe I'm being overly cautious but I think we should avoid it.

**Comment by kvlou** [comment]
@genslein thanks, moved the logging for users in the error log that goes to syslog

**Comment by kvlou** [comment]
[pgextwlist + credentials vuln remediation](https://trello.com/c/VoMqgNHJ/2248-pgextwlist-credentials-vuln-remediation)

**Pull request closed without merging**